### PR TITLE
Simplifies output from RAT Main

### PR DESCRIPTION
**Pull request by DrPaulSharp**

### CODE REVIEW

**Review by StephenNneji — COMMENTED** [review]
Good job with this one. See comments

File: API/makeEmptyResultStruct.m [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        fitNames{i} = dummyName;
+    end
+
+    result = struct('reflectivity', {reflectivity}, ...
+                    'simulation', {simulation}, ...
+                    'shiftedData', {shiftedData}, ...
+                    'layerSlds', {layerSlds}, ...
+                    'sldProfiles', {sldProfiles}, ...
+                    'allLayers', {allLayers}, ...
+                    'calculationResults', calculationResults, ...
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/makeEmptyResultStruct.m:164]: It's a legacy issue that I had meant to address and forgot. On it now!

File: API/makeEmptyResultStruct.m [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    end
+    
+    bestFitPars = zeros(1,nPars);
+    coder.varsize('bestFitPars',[1 10000],[0 1]);
+    
+    fitNames = cell(nPars,1);
+    coder.varsize('fitNames',[10000 1],[1 0]);   
+
+    dummyName = '';
+    coder.varsize('dummyName',[1 1000],[0 1]); % May not be necessary..
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/makeEmptyResultStruct.m:152]: Let's play and see.

File: minimisers/generalUtils/makeEmptyBayesResultsStruct.m [hunk 3]
```diff
@@ -91,16 +91,16 @@
     
     if isDomains
         sldXdata = cell(nContrasts,2);
-        sldDataCell2 = [1 1 1 ; 1 1 1];
-        coder.varsize('sldDataCell',[1 1e4],[1 1]); 
+        sldDataCell = [1 1 1; 1 1 1];
+        coder.varsize('sldDataCell',[2 1e4],[1 1]);
```
Comment (DrPaulSharp) [inline minimisers/generalUtils/makeEmptyBayesResultsStruct.m:95]: It was solving this bug (the use of "sldDataCell" and "sldDataCell2") that caused/exposed the other problems here.

**Pull request merged**